src/UCTNode.cpp

```diff
@@ -424,26 +424,32 @@ UCTNode::node_ptr_t UCTNode::find_new_root(const int move) {
 }
 
 // Use this version if the child could be anywhere.
-// Also updates the GameState.
 // Gives up after searching the direct children.
-UCTNode::node_ptr_t UCTNode::find_new_root(const GameState& g_new,
-                                           GameState& g_curr) {
-    if (m_has_children) {
-        for (auto& child : m_children) {
-            auto move = child->get_move();
-            if (g_new.get_last_move() == move) {
-                g_curr.play_move(move);
-                if (g_curr.board.get_hash() == g_new.board.get_hash()) {
-                    return std::move(child);
+// Also copies the GameState.
+void UCTNode::find_new_root(const GameState& g_new,
+                            GameState& g_curr,
+                            node_ptr_t& root) {
```

Comment:
Are there any cases where root wouldn't be this?

Reply:
Not sure what you're asking. This function used to return the new root, but I refactored it to include the case were the root is actually the same. I couldn't figure how to get it to return a pointer to itself, it was segfaulting.

So now UCTSearch passes it's current root pointer, and this function puts the new one in if needed.

Reply:
I mean the call is m_root->call(m_root) which suggests that m_root->call() is possible and inside you can do *this = blah.

Reply:
Aha, I will try that way tonight.